### ISSUE TITLE
Nuxt generate missing dependency `node-fetch@^2.5.0`

### ISSUE THREAD

**Issue by tmorehouse** [nuxt/nuxt.js#5724]
### Version

[v2.7.0](https://github.com/nuxt.js/releases/tag/v2.7.0)

### Reproduction link

[https://app.netlify.com/sites/bootstrap-vue/deploys/5cdab75508cf930008ed8019](https://app.netlify.com/sites/bootstrap-vue/deploys/5cdab75508cf930008ed8019)

### Steps to reproduce

Running nuxt generate produces the following error output (using 2.7.0 release)

```bash
9:41:58 AM: Executing user command: yarn docs-gen
9:41:59 AM: yarn run v1.3.2
9:41:59 AM: $ cross-env NODE_ENV=docs nuxt generate -c docs/nuxt.config.js
9:42:02 AM: 12:42:02 ℹ Production build
9:42:02 AM: 12:42:02  WARN  node-fetch@^2.5.0 is required but node-fetch@2.3.0 is installed!
9:42:02 AM: 12:42:02  ERROR  Please install missing dependencies:
9:42:02 AM:  Using yarn:
9:42:02 AM:  yarn add node-fetch@^2.5.0
9:42:02 AM:  Using npm:
9:42:02 AM:  npm i node-fetch@^2.5.0
9:42:02 AM: 12:42:02  FATAL  Missing App Dependencies
9:42:02 AM:   at Builder.validateTemplate (node_modules/@nuxt/builder/dist/builder.js:5645:13)
9:42:02 AM:   at Builder.build (node_modules/@nuxt/builder/dist/builder.js:5553:12)
9:42:02 AM:   at <anonymous>
9:42:02 AM:    ╭─────────────────────────────────────╮
9:42:02 AM:    │                                     │
9:42:02 AM:    │   ✖ Nuxt Fatal Error                │
9:42:02 AM:    │                                     │
9:42:02 AM:    │   Error: Missing App Dependencies   │
9:42:02 AM:    │                                     │
9:42:02 AM:    ╰─────────────────────────────────────╯
9:42:02 AM: error Command failed with exit code 1.
9:42:02 AM: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
``` 

### What is expected ?

Nuxt generate would work

### What is actually happening?

Fatal error missing correct dependency

### Additional comments?

You may need to update your dep versions

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c9210">#c9210</a>)</em></sub></div>

**Comment by pi0** [comment]
As the error says, the installed version of `node-fetch` is not recommended one and solution is `yarn add --dev node-fetch@^2.5.0`. Meanwhile, I've created #5723 to convert it to a warning.

**Comment by tmorehouse** [comment]
Although `node-fetch` is not one of our dependencies, so not sure where that one is coming from.

**Comment by pi0** [comment]
@tmorehouse unfortunately probably from a sub-dep hoisted by the package manager. Cleaning up `yarn.lock` prevents such issues.

**Comment by pi0** [comment]
shipped in [v2.7.1](https://github.com/nuxt/nuxt.js/releases/tag/2.7.1)
